make rostest in CMakeLists optional (ros/rosdistro#3010)

bulwahn opened the pull request.

Commit:
Allow multiple hw interfaces, Fix #112, and test.

- Allow to specify multiple hardware interfaces for joints and actuators.

- Fix invalid xml_element tag. Contents are now stored as a string.

- Unit test parser.

Commit:
Fix doc typo.

Commit:
Add class for holding transmission interfaces.

- Mirrors hardware_interface::RobotHW, but for transmissions.

Commit:
First draft of transmission loading.

- Only simple transmission type currently supported.

- Can load forward map for act->jnt state and jnt->act pos,vel.eff commands.

- Partial testing.

Commit:
Update loader test, better log statements.

Commit:
Remove dead code.

Commit:
Move common XML parsing code to TransmissionLoader

Mechanical reductions, offsets and roles are used by many transmission types.
The TransmissionLoader base class exposes convenience methods for parsing these
elements.

Commit:
Add differential drive transmission parser.

Commit:
Add four-bar-linkage transmission parser.

Commit:
Test greceful error-out with unsupported features.

Commit:
Log message change.

Commit:
Fix bug when adding multiple transmissions.

- std::vectors were being used to store raw joint data, and when new transmissions
  were added, push_back()s would (potentially) reallocate the vectors and
  invalidate already stored pointers in hardware_interfaces. We now use std::map.

- Move plugin implementations to a separate library.

- Export link libraries to the outside.

- More complete tests.

Commit:
Catkin fixes.

Commit:
Remove unnecessary header dependencies.

Commit:
Trivial test addition.

Commit:
Use less pointers in transmission loader data.

- Only RobotHW and RobotTransmission instances are pointers as they are owned
  by the robot hardware abstraction. The rest are plain members whose lifetime
  is bound to the loader struct.

Commit:
Test transmission handle duplication.

Commit:
Fix license header in some files.

Commit:
Fix compiler warning.

Commit:
Efficiency fix.

- cppcheck flagged a [passedByValue] warning. Using const references instead.

Commit:
Fix resource check for multi-dof transmisisons.

Commit:
C++11 compatibility fixes.

Commit:
Trivial, cosmetic fixes.

Commit:
Enable joint reduction spec for 4-bar linkages.

- As in the differential transmission, it's convenient to specify an additional
  mechanical reduction on the joint output. This is especially convenient for
  flipping the rotation direction of a joint (negative reduction value).

- Update URDF loader.

- Update documentation and tests.

Commit:
Perform sanity checks on members, not parameters.

- The result is the same, but this is more uniform with the rest of the code.

Commit:
Fix possible memory corruption in tests.

Commit:
Build script fixes.

- Add missing libraries to catkin_package call.

- Gate tests with CATKIN_ENABLE_TESTING.

- Add missing files to install target.

Commit:
Add developer documentation.

davetcoleman commented:
This seems to have broken the Travis builds, can you investigate? https://travis-ci.org/ros-controls/ros_control/builds/19742007

bulwahn commented:
Yes, I will investigate.

Commit:
Remove rosbuild artifacts. Fix #154.

Commit:
Merge pull request #155 from pal-robotics/indigo-devel

Remove rosbuild artifacts. Fix #154.

Commit:
Merge pull request #156 from pal-robotics/transmission-loader-indigo

Implement transmission loading from URDF - Indigo

Commit:
controller_manager_tests: fix library linking

From patch provided by po1 on hydro-devel.

davetcoleman committed:
Changed ros install to indigo

davetcoleman committed:
Merge pull request #159 from ros-controls/change_ros_install

Change ros install to indigo branches

davetcoleman committed:
Updated changelogs

davetcoleman committed:
0.8.0

Commit:
Register plugin under a group. Fixes #162.

Commit:
Merge pull request #163 from pal-robotics/rqt-cm-group

Register plugin under a group. Fixes #162.

Commit:
Use upstream liburdfdom-dev package.

Refs ros/rosdistro#4633.

Commit:
Merge pull request #168 from pal-robotics/liburdfdom-dev

Use upstream liburdfdom-dev package.

Commit:
Update changelogs.

Commit:
0.8.1

Commit:
Switch TravisCI to use indigo-devel.

davetcoleman committed:
Merge pull request #169 from pal-robotics/travis-indigo

Switch TravisCI to use indigo-devel.

Commit:
Restore hydro for CI.

Commit:
Merge pull request #170 from pal-robotics/travis-indigo

Restore hydro for CI.

Commit:
Propagate urdfdom changes to CMakeLists.txt

urdfdom is now standalone, so it must be find_package'd independently.
Also, the rosparam rostest was not being built correctly.

Commit:
Merge pull request #172 from pal-robotics/liburdfdom-dev

Propagate urdfdom changes to CMakeLists.txt

Commit:
Update changelogs

Commit:
0.8.2

Commit:
fix spelling errors

Commit:
Merge pull request #173 from shadowmanos/indigo-devel

Fix spelling errors

Commit:
make rostest in CMakeLists optional (ros/rosdistro#3010)

bulwahn commented:
Closed pull request, as original commit is outdated. I will re-open a new pull request.
